src/Jackalope/NodeType/NodeProcessor.php

```diff
@@ -71,18 +72,18 @@ class NodeProcessor
     private $autoLastModified;
 
     /**
-     * @var array List of namespaces known to the current session. Keys are prefix, values are URI.
+     * @var ArrayObject List of namespaces known to the current session. Keys are prefix, values are URI.
      */
-    private $namespaces = array();
+    private $namespaces;
 
     /**
-     * @param string  $userId           ID of the connected user
-     * @param array   $namespaces       List of namespaces in the current session. Keys are prefix, values are URI.
-     * @param boolean $autoLastModified Whether the last modified property should be updated automatically
+     * @param string      $userId           ID of the connected user
+     * @param ArrayObject $namespaces       List of namespaces in the current session. Keys are prefix, values are URI.
+     * @param bool        $autoLastModified Whether the last modified property should be updated automatically
      */
     public function __construct(
         $userId,
-        $namespaces = array(),
+        ArrayObject $namespaces,
```

Comment:
I tried to search where is NodeProcessor used and I think that is actually only in Jackalope Doctrine Dbal Client and NodeProcessorTest:

![snimek obrazovky 2017-01-20 v 14 54 46](https://cloud.githubusercontent.com/assets/7081340/22151926/9cb8386c-df20-11e6-9749-a41227e11310.png)

And there aren't namespaces optional so I don't know...